### PR TITLE
Fix IPFS review feedback

### PR DESCRIPTION
**Pull request by markg85**
- Fixed endianness bug in gateway file parsing
- Use IPFS_PATH in tests where IPFS_DATA was used
- Fixed typos from traling -> trailing
- Fixed broken link in IPFS.md

cc https://github.com/curl/curl/pull/12152

### CODE REVIEW

File: tests/data/test736 [hunk 1]
```diff
@@ -32,10 +32,10 @@ http
 </server>
 <setenv>
 HOME=%PWD/%LOGDIR
-IPFS_DATA=$HOME/.ipfs
+IPFS_PATH=$HOME/.ipfs
```
Comment (mkauf) [inline tests/data/test736:35]: To test that `IPFS_PATH` is really used by the curl tool, don't change `HOME`, and set `IPFS_PATH` to `%PWD/%LOGDIR/.ipfs`

(also in test737 and test738)

**Comment by mkauf** [comment]
The tests 736 and 737 now fail.

**Comment by markg85** [comment]
I'll look into it this evening.

File: tests/data/test736 [hunk 2]
```diff
@@ -32,7 +32,7 @@ http
 </server>
 <setenv>
 HOME=%PWD/%LOGDIR
-IPFS_PATH=$HOME/.ipfs
+IPFS_PATH=%HOME/.ipfs
```
Comment (bagder) [inline tests/data/test736:35]: `IPFS_PATH=%LOGDIR/.ipfs` is probably what you want.

File: tests/data/test736 (hunk 1 above)
Comment (markg85) [reply tests/data/test736]: Agreed, not setting HOME is preferred. Removed.

File: tests/data/test736 (hunk 2 above)
Comment (markg85) [reply tests/data/test736]: Yes, that looks like the intended behavior. Lets see what the build nodes think of it. Applied.

**Comment by bagder** [comment]
The broken markdown link is fixed here: https://github.com/curl/curl/pull/12308

**Comment by markg85** [comment]
> The broken markdown link is fixed here: https://github.com/curl/curl/pull/12308

Nice! Thank you for letting me know :)

Squashed as one commit. It's ready to be merged if there are no further objections.

**Review by bagder — APPROVED** [review]

**Pull request closed without merging**